sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Filter.java

```diff
@@ -0,0 +1,186 @@
+ * }</pre>
+ *
+ * Predicates that require examining multiple fields at once are also supported. For example,
+ * consider the following class representing a user account:
+ *
+ * <pre>{@code
+ * class UserAccount {
+ *   public double spendOnBooks;
+ *   public double spendOnMovies;
+ *       ...
+ * }
+ * }</pre>
+ *
+ * Say you want to examine only users whos total spend is above $100. You could write:
+ *
+ * <pre>{@code
+ * PCollection<UserAccount> users = readUsers();
+ * users.apply(Filter
+ *    .whereFieldNames(Lists.newArrayList("spendOnBooks", "spendOnMovies"),
+ *        row -> return row.getDouble("spendOnBooks") + row.getDouble("spendOnMovies") > 100.00));
+ * }</pre>
+ */
+@Experimental(Kind.SCHEMAS)
+public class Filter {
+  public static <T> Inner<T> create() {
+    return new Inner<T>();
+  }
+
+  /** Implementation of the filter. * */
+  public static class Inner<T> extends PTransform<PCollection<T>, PCollection<T>> {
+    Map<String, SerializableFunction<?, Boolean>> fieldNameFilters = Maps.newHashMap();
+    Map<Integer, SerializableFunction<?, Boolean>> fieldIdFilters = Maps.newHashMap();
+    Map<List<String>, SerializableFunction<Row, Boolean>> fieldNamesFilters = Maps.newHashMap();
+    Map<List<Integer>, SerializableFunction<Row, Boolean>> fieldIdsFilters = Maps.newHashMap();
+
+    /** Set a predicate based on the value of a field, where the field is specified by name. */
+    public Inner<T> whereFieldName(String fieldName, SerializableFunction<?, Boolean> predicate) {
+      fieldNameFilters.put(fieldName, predicate);
+      return this;
+    }
+
+    /** Set a predicate based on the value of a field, where the field is specified by id. */
+    public Inner<T> whereFieldId(int fieldId, SerializableFunction<?, Boolean> predicate) {
+      fieldIdFilters.put(fieldId, predicate);
+      return this;
+    }
+
+    /** Set a predicate based on the value of multipled fields, specified by name. */
+    public Inner<T> whereFieldNames(
+        List<String> fieldNames, SerializableFunction<Row, Boolean> predicate) {
+      fieldNamesFilters.put(fieldNames, predicate);
+      return this;
+    }
+
+    /** Set a predicate based on the value of multipled fields, specified by id. */
+    public Inner<T> whereFieldIds(
+        List<Integer> fieldIds, SerializableFunction<Row, Boolean> predicate) {
+      fieldIdsFilters.put(fieldIds, predicate);
+      return this;
+    }
+
+    @Override
+    public PCollection<T> expand(PCollection<T> input) {
+      // Validate that all referenced fields are in the schema.
+      Schema schema = input.getSchema();
+      for (String fieldName :
+          Sets.union(
+              fieldNameFilters.keySet(),
+              fieldNamesFilters
+                  .keySet()
+                  .stream()
+                  .flatMap(List::stream)
+                  .collect(Collectors.toSet()))) {
+        schema.getField(fieldName);
```

Comment:
This appears to work because getField throws an exception. Can you add a test for this case?

Reply:
added tests to both FilterTest and SelectTest